### ISSUE TITLE
[1.4.2] Server doesn't get styles generated by injectGlobal

### ISSUE THREAD

**Issue by diegohaz** [styled-components/styled-components#423]
As now we are using another `StyleSheet` object to generate `injectGlobal` styles, the following method doesn't work properly anymore:
```js
import styleSheet from 'styled-components/lib/models/StyleSheet'
...
const styles = styleSheet.rules().map(rule => rule.cssText).join('\n')
```

Now we need also to import and get rules from `GlobalStyleSheet`
```js
import styleSheet from 'styled-components/lib/models/StyleSheet'
import globalStyleSheet from 'styled-components/lib/models/GlobalStyleSheet'
...
const styles = styleSheet.rules().map(rule => rule.cssText).join('\n')
const globalStyles = globalStyleSheet.rules().map(rule => rule.cssText).join('\n')
```

That works, but as some projects are already counting on the first method and we [encouraged that](https://github.com/styled-components/styled-components/pull/214#issuecomment-267519606), I think we should return global styles in `StyleSheet`.

The good news is that 1.4.2 also fixes #238 🎉

**Comment by diegohaz** [comment]
I can also confirm that concatenating `globalStyles + styles` doesn't break #238

**Comment by mxstbr** [comment]
Ah damn, I totally forgot we'd break this use case! My bad, mind submitting a PR reverting that change and a new PR against the `v2` branch with those changes?!

**Comment by diegohaz** [comment]
If no one else takes it first, I can work on this on Tuesday. But I would try to keep the changes from `1.4.2` in `v1`. It's too good and solve big issues in `v1` to be left to `v2` 😣

**Comment by mxstbr** [comment]
> But I would try to keep the changes from 1.4.2 in v1. It's too good and solve big issues in v1 to be left to v2 😣

It's a breaking change and is thusly going to hurt a lot of people, we can't do that I'm afraid 😕

**Comment by xcoderzach** [comment]
@diegohaz @mxstbr Sorry about breaking this 😦, totally didn't think about this case.  What properties of the StyleSheet module are "public"?  If it's just `.rules()`, we can definitely make this work with `v1`.  I'll put something together tonight.

**Comment by xcoderzach** [comment]
My plan is to revert to a single StyleSheet module, which handles both global and component styles.  I'd add an global option argument (i.e. `{ global: true }`)  to the `.insert` method which would decide which sheet to insert into.  Then the rules method would just return the concatenation of `globalSheet.rules()` and `componentSheet.rules()`

**Comment by diegohaz** [comment]
@xcoderzach No problem. We should had tested `.rules()` properly to make sure it wouldn't break. I love your idea. It would be awesome to have it working in `v1`. 🎉

**Comment by mxstbr** [comment]
That sounds like a nice solution @xcoderzach!

**Comment by xcoderzach** [comment]
Should be fixed in #440

**Comment by mxstbr** [comment]
This is fixed in v2.
